internal/brokers/examplebroker/broker.go

```diff
@@ -110,6 +169,58 @@ func (b *Broker) GetAuthenticationModes(ctx context.Context, sessionID string, s
 	}
 
 	//var candidatesAuthenticationModes []map[string]string
+	allModes := getSupportedModes(sessionInfo, supportedUILayouts)
+
+	// If the user needs mfa, we remove the last used mode from the list of available modes.
+	if sessionInfo.currentMfaStep > 0 && sessionInfo.currentMfaStep < sessionInfo.neededMfa {
+		allModes = getMfaModes(sessionInfo, sessionInfo.allModes)
+	}
+	// If the user needs or can reset the password, we only show those authentication modes.
```

Comment:
it should rather append those authentication mode you mean? (It’s after a first successful authentication, as the last step)

Reply:
Yes, actually, we should not append them, see my other comments down the lines in the IsAuthorized part :)